Sources/Send/Input/Chain/SendInputBusinessLogic+ValidationChain.swift

```diff
@@ -50,6 +50,18 @@ extension SendInputBusinessLogic {
         if state.feeInToken.total > (wallet.lamports ?? 0) {
             return state.copy(status: .error(reason: .insufficientAmountToCoverFee))
         }
+
+        if let context = state.feeRelayerContext, wallet.isNativeSOL {
+            // Separate logic for fee paid with SOL. Need to consider minimumRelayAccountBalance
+            let leftAmount = (Int64(wallet.lamports ?? 0) - Int64(state.feeInToken.total))
```

Comment:
why are you casting from UInt64 to Int64 and then compare?
wallet.lamports -> UInt64
state.feeInToken.total -> UInt64
context.minimumRelayAccountBalance -> UInt64

Reply:
to handle negative numbers